app/src/handle-request.lisp

```diff
@@ -166,4 +149,38 @@ The mapping vector V specifies that the qubit as specified in the program V[i] h
                  (qvm:map-amplitudes
                   qvm
                   (lambda (z) (write-complex-double-float-as-binary z reply-stream)))))
-             (format-log "Response sent in ~D ms." send-response-time))))))))
+             (format-log "Response sent in ~D ms." send-response-time))))
+        
+        ((:probabilities)
+         (check-for-quil-instrs-field js)
+         (let* ((isns (get-quil-instrs-field js))
+                (quil (let ((quil:*allow-unresolved-applications* t))
+                        (process-quil (safely-parse-quil-string isns))))
+                (num-qubits (cl-quil:qubits-needed quil)))
+           (let (send-response-time)
+             (multiple-value-bind (qvm probabilities)
+                 (perform-probabilities *simulation-method* quil num-qubits
+                                        :gate-noise gate-noise
+                                        :measurement-noise measurement-noise)
+               (with-timing (send-response-time)
+                 (setf (tbnl:content-type*) "application/octet-stream")
+                 (setf (tbnl:header-out ':ACCEPT) "application/octet-stream")
+                 (setf (tbnl:content-length*)
+                       (* qvm::+octets-per-flonum+ (length probabilities)))
+                 (let ((reply-stream (tbnl:send-headers)))
+                   (map nil
+                        (lambda (x) (write-double-float-as-binary x reply-stream))
+                        probabilities))))
+             (format-log "Response sent in ~D ms." send-response-time))))
+
+        ((:run-for-effect)
+         (check-for-quil-instrs-field js)
+         (let* ((isns (get-quil-instrs-field js))
+                (quil (let ((quil:*allow-unresolved-applications* t))
+                        (process-quil (safely-parse-quil-string isns))))
+                (num-qubits (cl-quil:qubits-needed quil)))
+           (perform-run-for-effect *simulation-method* quil num-qubits
+                                   :gate-noise gate-noise
+                                   :measurement-noise measurement-noise)
+           (with-output-to-string (s)
```

Comment:
Since this will be constant, you can wrap it in `(load-time-value ...)`, or just return `"true"` if you're devious.